### PR TITLE
feat(EMS-868-885): Account sign in - request a new security code, success banner

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR adds the ability to request a new security code during the account sign in process.

## Changes

- Create new page template, controller and route, If API calls are successful, add a flag to `req.flash` for the next part of the flow.
- Add new UI GQL API call, `signInSendNewCode`.
- Add new GQL resolver, `accountSignInSendNewCode`. This accepts an accountId, check it exists and if so, generate a new OTP and email it to the user.
- Update the "enter code" page/controlelr to check if `req.flash` has a "new code sent "flag. If so, render a success banner.
- Add E2E test coverage for the "request new code" page, and the full journey i.e submitting this form, success banner appears and a user can sign in with the new security code.

## Other improvements
- Delete exporter by ID after running account E2E tests with new cypress command `cy.deleteExporter`. This should stop some tests from randomly failing RE account verification/unverified tests.
- Update API tests to use a new generic mock of OTP response.
- Rename E2E API call function from`addAndGetOtp` to `addAndGetOTP`.
- FIxed a failing UI test for contract completion date due to the date (was failing when day of the month is the first day).
- Fixed some typos.

### CODE REVIEW

File: e2e-tests/cypress/e2e/journeys/insurance/account/sign-in/enter-code/validation/account-sign-in-enter-code-validation.spec.js [hunk 1]
```diff
@@ -69,7 +70,7 @@ context('Insurance - Account - Sign in - Enter code - validation', () => {
     });
 
     describe('when submitting a valid security code', () => {
-      it(`should redirect to ${DASHBOARD}`, () => {
+      it(`should redirect to ${ROOT}${DASHBOARD}`, () => {
```
Comment (Zainzzkk) [inline e2e-tests/cypress/e2e/journeys/insurance/account/sign-in/enter-code/validation/account-sign-in-enter-code-validation.spec.js:73]: what if the dashboard in routes had DASHBOARD: ${ROOT}${DASHBOARD} then would not need to add it here?

Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/account/sign-in/enter-code/validation/account-sign-in-enter-code-validation.spec.js]: Great shout, this was the case in UI constants, but E2E was missed in the update. Fixed

File: src/ui/server/controllers/insurance/account/sign-in/request-new-code/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      });
+
+      it(`should redirect to ${SIGN_IN_ROOT}`, async () => {
+        await post(req, res);
+
+        expect(res.redirect).toHaveBeenCalledWith(SIGN_IN_ROOT);
+      });
+    });
+
+    it('should call api.keystone.account.verifyAccountSignInCode', async () => {
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/account/sign-in/request-new-code/index.test.ts:88]: can these be wrapped in their own describe

File: src/api/helpers/generate-otp-and-update-account.test.ts [hunk 3]
```diff
@@ -56,16 +49,16 @@ describe('helpers/generate-otp-and-update-account', () => {
   });
 
   test('it should generate an OTP and save to the account', async () => {
-    expect(account.otpSalt).toEqual(mockOtp.salt);
-    expect(account.otpHash).toEqual(mockOtp.hash);
+    expect(account.otpSalt).toEqual(mockOTP.salt);
+    expect(account.otpHash).toEqual(mockOTP.hash);
     // @ts-ignore
```
[the remark this answers is not in the document]
Comment (ttbarnes) [reply src/api/helpers/generate-otp-and-update-account.test.ts:54]: No, not sure why it was in there. Removed

**Review by Zainzzkk — DISMISSED** [review]
LGTM

**Review by Zainzzkk — DISMISSED** [review]
lgtm

**Pull request merged**